packages/client/lib/client/index.ts

```diff
@@ -239,19 +263,14 @@ export default class RedisClient<
     }
 
     #initiateOptions(options?: RedisClientOptions<M, F, S>): RedisClientOptions<M, F, S> | undefined {
-        if (options?.url) {
-            const parsed = RedisClient.parseURL(options.url);
-            if (options.socket) {
-                parsed.socket = Object.assign(options.socket, parsed.socket);
-            }
-
-            Object.assign(options, parsed);
-        }
-
         if (options?.database) {
             this.#selectedDB = options.database;
         }
 
+        if (options) {
+            return RedisClient.parseOptions(options);
```

Comment:
maybe better to put before `this.#selectedDB = options.database` but it's a private method use in a constructor anyways?